Sources/Support/StoreMessagesHelper.swift

```diff
@@ -57,16 +57,22 @@ actor StoreMessagesHelper: StoreMessagesHelperType {
     }
 
     func showStoreMessages(types: Set<StoreMessageType>) async {
+        var displayedMessages: [StoreMessage] = []
         for message in self.deferredMessages {
             if let messageType = message.reason.messageType, types.contains(messageType) {
                 do {
                     try await message.display(in: self.systemInfo.currentWindowScene)
+                    displayedMessages.append(message)
                 } catch {
                     Logger.error(Strings.storeKit.error_displaying_store_message(error))
                 }
             }
         }
-        self.deferredMessages.removeAll()
+        self.deferredMessages.removeAll { message in
+            displayedMessages.contains { displayedMessage in
+                displayedMessage.reason == message.reason
+            }
+        }
```

Comment:
Maybe `displayedMessages` could be a set (of `reason`s) to avoid having to iterate over it for every item in `deferredMessages`?

(Just for my understanding: why are we comparing on `reason`? I noticed that a `Message` also has a [`hashValue`](https://developer.apple.com/documentation/storekit/message/3954509-hashvalue).)

Reply:
Great question! I was chatting with @MarkVillacampa about this on Friday. We were comparing on `reason` because the `StoreMessage` abstraction only exposed `reason`, and that was all we had available. After talking it through with @MarkVillacampa, we decided to add and compare a hashValue, so it'll only remove the exact instance that was shown. Great catch! :D

Reply:
Nice, thanks for explaining! And I'm happy that it was a useful question haha.